DJHelper/DJHelper/Model/Event Model/Event Controller/EventController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  EventController.swift
+//  DJHelper
+//
+//  Created by Michael Flowers on 6/1/20.
+//  Copyright © 2020 craigswanson. All rights reserved.
+//
+
+import Foundation
+class EventController {
+    enum EventErrors: Error {
+        case authorizationError(Error)
+        case noDataError
+        case encodeError(Error)
+        case decodeError(Error)
+        case otherError(Error)
+    }
+    private let baseURL = URL(string: "https://api.dj-helper.com/api/auth/event/")!
+    let dataLoader: NetworkDataLoader
+    
+    init(dataLoader: NetworkDataLoader = URLSession.shared){
+        self.dataLoader = dataLoader
+    }
+    
+    //MARK: - AUTHORIZE AN EVENT
+    ///The server returns an object with the event data
+    func authorize(event: Event, completion: @escaping (Result<EventRepresentation, EventErrors>) -> Void){
+        guard let eventToAuthorize = event.eventAuthorizationRep else { return }
+//        guard let eventAuthRequest = event.eventAuthRequest else { return }
+        
+        var urlRequest = URLRequest(url: baseURL)
+        urlRequest.httpMethod = HTTPMethod.post.rawValue
+        urlRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        let encoder = JSONEncoder()
+        encoder.dateEncodingStrategy = .iso8601
+//        encoder.keyEncodingStrategy = .convertToSnakeCase
```

Comment:
I'm not sure why the .convertToSnakeCase strategy is commented out for the encoder?  Are we using the coding keys here?

Reply:
I left that in there because I was unsure if you wanted to use the custom coding keys or my Request model. 

Its currently commented out because we are using the custom coding keys. 

If we use the request model struct EventAuthRequest, then we would uncomment the snakeCase and use it.